types/src/get_with_proof.rs

```diff
@@ -192,6 +192,11 @@ pub fn verify_update_to_latest_ledger_response<'a>(
         req_client_known_version,
     );
 
+    // Verify any validator change proof and latest ledger info. Provide an
+    // updated trusted state if the proof is correct and not stale.
+    let trusted_state_change =
+        trusted_state.verify_and_ratchet(ledger_info_with_sigs, validator_change_proof)?;
```

Comment:
when the li returned is stale, does it make any sense to proceed verifying the items? (because we only know it's stale by looking at a smaller version, but we don't know if it is a fork from an old version, right?)

What is the user of resp.verify() supposed to do with a "stale"? I think maybe it's better to raise an error to keep the user from making mistakes?

Reply:
The intention was to separate a malicious error (server returns bogus proofs) from a stale message (which could happen if you send multiple requests concurrently between different servers while ratcheting the same `TrustedState`). But you're right, I don't think users should ever use a stale response, so it should be an `Err` : )

Reply:
maybe this motivates some more specific error types eventually?